target/linux/ramips/mt7620/config-5.4

```diff
@@ -174,6 +174,7 @@ CONFIG_MTD_SPLIT_JIMAGE_FW=y
 CONFIG_MTD_SPLIT_SEAMA_FW=y
 CONFIG_MTD_SPLIT_TPLINK_FW=y
 CONFIG_MTD_SPLIT_UIMAGE_FW=y
+CONFIG_MTD_VIRT_CONCAT=y
```

Comment:
Would be interesting to know what's the size requirement here, on a constrained target like mt7620 ...

Reply:
What file (and where) should one check after building an image with and without this change?

Reply:
I can build and check, no issue. How to "force" the build to 5.4 (as it's defaulting to 4.14.187 right now)?

Reply:
mt7620 has been updated to 5.4 by default quite a while ago. If you still have 4.14 by default, you should rebase to a recent master.

Reply:
I figured that was the answer, and was afraid you'd say that ... LOL! OK, let me try that - rebase is definitely not my friend 😞.

Reply:
But also then ... is there a way to check the delta you're looking for above? Just try with and without that CONFIG set? And what file to check after ... kernel size?

Reply:
Woohoo! Did a rebase, it was clean - make clean, then build and flash => came back up, all settings saved, now on 5.4.60. Life is good 😆. What do you want me to check (i.e. size-wise)?

Thanks!

Reply:
FYI, checked - just to make sure! The file in my build now,
CONFIG_MTD_VIRT_CONCAT=y

Reply:
Hi. Marking this as resolved, as it seems to be running great (all my Windows Desktop PC traffic is going though it, for 2 days now ... and no issues at all. Yay!). But if something is needed, please don't hesitate - just let me know. Trying to resolve these if possible, to handle the commit comment below. Thanks!!

Reply:
Why are you marking this as resolved? I didn't ask whether it was "running great", but what the size impact was.

Reply:
Sorry - it was because I asked above what size you were after, didn't hear back. I (incorrectly!) assumed that if it was running you weren't so concerned about it any more. My mistake 😞.

Do you mean the kernel size? No issue at all here pulling the info, just not sure what you're after. Thanks!

Reply:
Well, the button is "resolved", not "closed". Things that are not "resolved" should stay open until the end.

You change a kernel symbol here, which will increase kernel size for _all_ devices in that subtarget, so it would indeed by interesting whether it will increase substantially and maybe break other devices.

Reply:
> Well, the button is "resolved", not "closed". Things that are not "resolved" should stay open until the end.

OK, that's my misunderstanding - sorry. I was assuming the intention was to address the noted items, then mark them as closed. Again, new to that part of things, wasn't meaning to break any rules, just didn't know how that's to be handled.

> You change a kernel symbol here, which will increase kernel size for all devices in that subtarget, so it would indeed by interesting whether it will increase substantially and maybe break other devices.

Ahhh, OK - get your point now. So try a build with and without this? I can do that, NP - just making sure 😉. Thanks!

Reply:
OK, I ran binwalk on the output (loader unchanged of course, currently 3292 bytes), sizes below - please just yell if you need something else! I do think some of these are unchanged (or minor change) due to padding (i.e. delta is "hidden" by this),
1. openwrt-ramips-mt7620-hootoo_ht-tm05-squashfs-sysupgrade.bin (Squashfs filesystem)
a) With CONFIG_MTD_VIRT_CONCAT=y, size: 3098418 bytes
b) # CONFIG_MTD_VIRT_CONCAT is not set, size: 3098402 bytes
2. openwrt-ramips-mt7620-hootoo_ht-tm05-squashfs-rootfs.bin
a) With CONFIG_MTD_VIRT_CONCAT=y, uncompressed size: 6112035 bytes
b) # CONFIG_MTD_VIRT_CONCAT is not set, uncompressed size: 6112035 bytes

Is that what you're after?

Thanks!

Reply:
@arrmo the concern is all about the kernel

Reply:
> is the kernel in the first file?

I believe it is - it's a bit "odd" now, as the kernel.bin is of course the lzma-loader. To make sure, is there a specific file to check (perhaps in build_dir)?

Thanks!

Reply:
Commit commit 976d3e22349c9da607777ad364c8639b5c531e04  + this PR + `CONFIG_ALL_KMODS=y`:
```
$ ls -gG build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux*
-rwxr-xr-x 1  6562620 Aug 31 12:53 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux
-rwxr-xr-x 1 39854480 Aug 31 12:53 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux.debug
-rwxr-xr-x 1  6567632 Aug 31 12:53 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux.elf
-rwxr-xr-x 1  8984700 Aug 31 12:54 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux-initramfs
-rwxr-xr-x 1 42290756 Aug 31 12:54 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux-initramfs.debug
-rwxr-xr-x 1  8989712 Aug 31 12:54 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux-initramfs.elf
```

Reply:
Commit commit 976d3e2 + this PR + `CONFIG_ALL_KMODS=y` + without `CONFIG_MTD_VIRT_CONCAT=y`:
```
$ ls -gG build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux*
-rwxr-xr-x 1  6562588 Aug 31 13:41 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux
-rwxr-xr-x 1 39845588 Aug 31 13:41 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux.debug
-rwxr-xr-x 1  6567600 Aug 31 13:41 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux.elf
-rwxr-xr-x 1  8984836 Aug 31 13:42 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux-initramfs
-rwxr-xr-x 1 42282028 Aug 31 13:42 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux-initramfs.debug
-rwxr-xr-x 1  8989848 Aug 31 13:42 build_dir/target-mipsel_24kc_musl/linux-ramips_mt7620/vmlinux-initramfs.elf
```

Reply:
file | with `CONCAT` | without `CONCAT` | change (without -> with)
--------|-----------|------------|----------
vmlinux | 6562620 | 6562588 | 32
vmlinux.debug | 39854480 | 39845588 |  8892
vmlinux.elf | 6567632 | 6567600 | 32
vmlinux-initramfs | 8984700 | 8984836 | -136
vmlinux-initramfs.debug | 42290756 | 42282028 | 8728
vmlinux-initramfs.elf | 8989712 | 8989848 | -136

Reply:
Thanks!